### ISSUE TITLE
Sign up error not finding user pool client

### ISSUE THREAD

**Issue by jmettes** [jagregory/cognito-local#19]
I'm not able to sign up a user to the user pool, I get resource not found. Any idea what's going on?

```
$ mkdir .cognito && echo '{}' > .cognito/config.json
```
```
$ cognito-local &
[1] 1038018
$ Cognito Local running on http://127.0.0.1:9229
```
```
$ aws cognito-idp create-user-pool-client --user-pool-id local --client-name client --endpoint http://127.0.0.1:9229
{
    "UserPoolClient": {
        "UserPoolId": "local",
        "ClientName": "client",
        "ClientId": "6o0d2h80xj7loqagfenro26c2",
        "LastModifiedDate": 1620798438504,
        "CreationDate": 1620798438504,
        "RefreshTokenValidity": 30,
        "AllowedOAuthFlowsUserPoolClient": false
    }
}
```
```
$ cat .cognito/db/clients.json 
{
  "Clients": {},
  "Clients,6o0d2h80xj7loqagfenro26c2": {
    "ClientId": "6o0d2h80xj7loqagfenro26c2",
    "ClientName": "client",
    "UserPoolId": "local",
    "CreationDate": 1620798438504,
    "LastModifiedDate": 1620798438504,
    "AllowedOAuthFlowsUserPoolClient": false,
    "RefreshTokenValidity": 30
  }
}
```
```
$ aws cognito-idp sign-up --client-id 6o0d2h80xj7loqagfenro26c2 --username username --password password --endpoint http://0.0.0.0:9229
Error handling target: SignUp ResourceNotFoundError: Resource not found
    at Object.getUserPoolForClientId (/home/jonathan/cognito-local/node_modules/cognito-local/lib/services/cognitoClient.js:25:15)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/targets/signUp.js:23:20
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/server/server.js:69:22 {
  code: 'CognitoLocal#ResourceNotFoundException'
}

An error occurred (400) when calling the SignUp operation: Resource not found
```
I thought maybe the format of that `.cognito/db/clients.json` looked wrong, so I changed it
```
$ cat .cognito/db/clients.json 
{
  "Clients": {
      "6o0d2h80xj7loqagfenro26c2": {
        "ClientId": "6o0d2h80xj7loqagfenro26c2",
        "ClientName": "client",
        "UserPoolId": "local",
        "CreationDate": 1620798438504,
        "LastModifiedDate": 1620798438504,
        "AllowedOAuthFlowsUserPoolClient": false,
        "RefreshTokenValidity": 30
      }
  }
}
```
but then I get a different error
```
$ aws cognito-idp sign-up --client-id 6o0d2h80xj7loqagfenro26c2 --username username --password password --endpoint http://0.0.0.0:9229
Error handling target: SignUp TypeError: Cannot read property 'filter' of undefined
    at /home/jonathan/cognito-local/node_modules/cognito-local/lib/targets/signUp.js:42:34
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/server/server.js:69:22
Error handling target: SignUp TypeError: Cannot read property 'filter' of undefined
    at /home/jonathan/cognito-local/node_modules/cognito-local/lib/targets/signUp.js:42:34
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/server/server.js:69:22
Error handling target: SignUp TypeError: Cannot read property 'filter' of undefined
    at /home/jonathan/cognito-local/node_modules/cognito-local/lib/targets/signUp.js:42:34
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/server/server.js:69:22
Error handling target: SignUp TypeError: Cannot read property 'filter' of undefined
    at /home/jonathan/cognito-local/node_modules/cognito-local/lib/targets/signUp.js:42:34
    at async /home/jonathan/cognito-local/node_modules/cognito-local/lib/server/server.js:69:22
^C
```

**Comment by jagregory** [comment]
Hey @jmettes, apologies for the _extremely_ slow response. That config file key in your cat of `clients.json` looks very suspicious. I suspect it's related to this PR: https://github.com/jagregory/cognito-local/pull/8

I'll get that one reviewed and hopefully it'll resolve this issue. I'm guessing I also need some more tests covering this, as it's a really obvious regression.

**Comment by jagregory** [comment]
Fix release in [v2.3.0](https://github.com/jagregory/cognito-local/releases/tag/v2.3.0)
